File: python/mod_8/task_4.py
def main():
    print("=====Убийца Steam=====")
    while True:
        print("Ввод:")
        file_size = float(input("Укажите размер файла для скачивания: "))
        connection_speed = float(input("Какова скорость вашего соединения? "))

        print("Вывод: ")
        seconds = 0
        downloaded = 0

        while downloaded < file_size:
            seconds += 1
            downloaded += connection_speed
            if downloaded > file_size:
                downloaded = file_size
            print(f"За {seconds} сек. Скачано {downloaded} из {file_size} МБ ({int((downloaded/file_size)*100)}%)")

if __name__ == "__main__":
    main()

File: final/mod_8/task_4.cpp
#include <iostream>

int main(){
    float fileSize, connectionSpeed, downloaded = 0;
    using namespace std;
    cout << "=====Убийца Steam=====\n";

    while (true){
        cout << "Ввод:\n";
        cout << "Укажите размер файла для скачивания: "; cin >> fileSize;
        cout << "Какова скорость вашего соединения? "; cin >> connectionSpeed;

        cout << "Вывод:\n";
        for(int sec=1; downloaded < fileSize; sec++) {
            downloaded += connectionSpeed;
            if (downloaded > fileSize) downloaded = fileSize;
            cout << "Прошло " << sec << " сек. Скачано "<< downloaded << " из " << fileSize << " МБ (" << (int)(downloaded/fileSize*100) << "%)\n";
            cout << "==========\n";
        }
    }


}